### ISSUE TITLE
Pylint: instance of Service has no `status` member

### ISSUE THREAD

**Issue by swgillespie** [pulumi/pulumi-kubernetes#503]
Because we don't annotate the properties that each object has on the Python class, pylint often rejects legitimate programs. We need to annotate the set of properties exported by each resource.

**Comment by lukehoban** [comment]
Do we have a repro of the issue in question here?

**Comment by lblackstone** [comment]
I don't have a repro, and I'm not completely sure what Sean was talking about. We don't include type hints on the generated Python SDK resource classes, so perhaps that was it?

For example, the following class doesn't type the constructor params, although it does check them inside the class. I'm not sure if there was a technical reason that this is the case.
```py3
class Deployment(pulumi.CustomResource):
    """
    Deployment enables declarative updates for Pods and ReplicaSets.
    """
    def __init__(self, resource_name, opts=None, metadata=None, spec=None, status=None, __name__=None, __opts__=None):
        if __name__ is not None:
            warnings.warn("explicit use of __name__ is deprecated", DeprecationWarning)
            resource_name = __name__
        if __opts__ is not None:
            warnings.warn("explicit use of __opts__ is deprecated, use 'opts' instead", DeprecationWarning)
            opts = __opts__
        if not resource_name:
            raise TypeError('Missing resource name argument (for URN creation)')
        if not isinstance(resource_name, str):
            raise TypeError('Expected resource name to be a string')
        if opts and not isinstance(opts, pulumi.ResourceOptions):
            raise TypeError('Expected resource options to be a ResourceOptions instance')

        __props__ = dict()

        __props__['apiVersion'] = 'apps/v1'
        __props__['kind'] = 'Deployment'
        __props__['metadata'] = metadata
        __props__['spec'] = spec
        __props__['status'] = status

        if opts is None:
            opts = pulumi.ResourceOptions()
        if opts.version is None:
            opts.version = version.get_version()

        super(Deployment, self).__init__(
            "kubernetes:apps/v1:Deployment",
            resource_name,
            __props__,
            opts)

    def translate_output_property(self, prop: str) -> str:
        return tables._CASING_FORWARD_TABLE.get(prop) or prop

    def translate_input_property(self, prop: str) -> str:
        return tables._CASING_BACKWARD_TABLE.get(prop) or prop
```

**Comment by lblackstone** [comment]
I'm guessing he was referring to docs like the following:
https://github.com/pulumi/pulumi-gcp/blob/cdc37abcda0659a15ae2914e71da2d79c2d1d7cf/sdk/python/pulumi_gcp/storage/bucket.py#L12-L73

**Comment by lukehoban** [comment]
Yes - we should be adding those. They are presumably also necessary for useful documentation.

**Comment by lblackstone** [comment]
Bumping to M26

**Comment by lblackstone** [comment]
I think this was fixed in #635
